billing-report-agent/src/main/java/com/epam/pipeline/billingreportagent/service/impl/converter/AwsStoragePriceListLoader.java

```diff
@@ -0,0 +1,153 @@
+
+    public AwsStoragePriceListLoader(final String awsStorageServiceName) {
+        this.awsStorageServiceName = awsStorageServiceName;
+        this.mapper = new ObjectMapper();
+    }
+
+    @Override
+    public Map<String, StoragePricing> loadFullPriceList() {
+        final Map<String, StoragePricing> fullPriceList = new HashMap<>();
+        loadAwsPricingCards(awsStorageServiceName).forEach(price -> {
+            final String regionName = price.getProduct().getAttributes().get(LOCATION_KEY);
+            getRegionFromFullLocation(regionName)
+                .ifPresent(region -> fullPriceList.put(region.getName(),
+                                                       convertAwsPricing(price.getTerms().getOnDemand())));
+        });
+        return fullPriceList;
+    }
+
+    @Override
+    public CloudProvider getProvider() {
+        return CloudProvider.AWS;
+    }
+
+    private StoragePricing convertAwsPricing(final Map<String, AwsPriceDimensions> allPrices) {
+        final StoragePricing pricing = new StoragePricing();
+        final List<AwsPriceRate> rates = CollectionUtils.emptyIfNull(allPrices.values()).stream()
+            .map(AwsPriceDimensions::getPriceDimensions)
+            .map(Map::values)
+            .flatMap(Collection::stream)
+            .collect(Collectors.toList());
+        rates.forEach(rate -> {
+            final BigDecimal priceGb = new BigDecimal(rate.getPricePerUnit().get(US_DOLLAR_CODE),
+                                                      new MathContext(PRECISION))
+                .multiply(BigDecimal.valueOf(CENTS_IN_DOLLAR));
+            final Long beginRange = rate.getBeginRange() * BYTES_TO_GB;
+            final Long endRange = rate.getEndRange().equals(Long.MAX_VALUE)
+                                  ? Long.MAX_VALUE
+                                  : rate.getEndRange() * BYTES_TO_GB;
+            final StoragePricing.StoragePricingEntity pricingEntity =
+                new StoragePricing.StoragePricingEntity(beginRange, endRange, priceGb);
+            pricing.addPrice(pricingEntity);
+        });
+        return pricing;
+    }
+
+    private List<AwsPricingCard> loadAwsPricingCards(final String awsStorageServiceName) {
+        final List<AwsPricingCard> allPrices = new ArrayList<>();
+        final Filter filter = new Filter();
+        filter.setType(TERM_MATCH_FILTER);
+        filter.setField(PRODUCT_FAMILY_KEY);
+        filter.setValue(STORAGE);
+        filter.setField(STORAGE_CLASS_KEY);
+        filter.setValue(GENERAL_STORAGE);
+
+        String nextToken = StringUtils.EMPTY;
+        do {
+            final GetProductsRequest request = new GetProductsRequest()
+                .withServiceCode(awsStorageServiceName)
+                .withFilters(filter)
+                .withNextToken(nextToken)
+                .withFormatVersion(AWS_PRICE_FORMAT_VERSION);
+
+            final AWSPricing awsPricingService = AWSPricingClientBuilder
```

Comment:
I think, we can init `awsPricingService ` once before cycle

Reply:
Done